code/modules/items/gimmick/figurine.dm

```diff
@@ -148,6 +158,11 @@
 		else
 			return ..()
 
+proc/add_to_donator_list(var/potential_donator_ckey)
+	for (var/datum/figure_info/patreon/fig as anything in concrete_typesof(/datum/figure_info/patreon))
```

Comment:
That's quite a long list to be iterating in `client/New`, maybe generate a list of donator ckeys on roundstart and then check if the ckey is in that list each time instead.